engine/crates/runtime-local/src/fetch.rs

```diff
@@ -10,28 +10,49 @@ use self::websockets::StreamingRequest;
 
 pub struct NativeFetcher {
     client: reqwest::Client,
+    /// Timeout configuration per subgraph
+    timeouts: HashMap<String, Duration>,
 }
 
 impl NativeFetcher {
     pub fn runtime_fetcher() -> Fetcher {
         Fetcher::new(Self {
             client: reqwest::Client::new(),
+            timeouts: HashMap::new(),
+        })
+    }
+
+    pub fn runtime_fetcher_with_timeouts(timeouts: HashMap<String, Duration>) -> Fetcher {
+        Fetcher::new(Self {
+            client: reqwest::Client::new(),
+            timeouts,
         })
     }
 }
 
 #[async_trait::async_trait]
 impl FetcherInner for NativeFetcher {
     async fn post(&self, request: FetchRequest<'_>) -> FetchResult<FetchResponse> {
-        let response = self
+        let subgraph_name = request.subgraph_name;
+
+        let mut http_request = self
             .client
             .post(request.url.clone())
             .body(request.json_body)
             .header("Content-Type", "application/json")
-            .headers(request.headers.clone())
-            .send()
-            .await
-            .map_err(|e| FetchError::AnyError(e.to_string()))?;
+            .headers(request.headers.clone());
+
+        if let Some(timeout) = self.timeouts.get(subgraph_name) {
```

Comment:
That fallback behaviour is wrong. There should be a default (from the config), not fall back to no timeout.

Reply:
That would be solved if we moved timeouts inside the engine's subgraph configuration instead. Is that the right move?